internal/app_config.go

```diff
@@ -0,0 +1,127 @@
+package internal
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/pelletier/go-toml"
+)
+
+var DEF_CONFIG_ORDER = []string{
+	"/usr/lib/sota/conf.d",
+	"/var/sota/sota.toml",
+	"/etc/sota/conf.d/",
+}
+
+type cfgFile struct {
+	name string
+	path string
+
+	tree *toml.Tree
+}
+
+type AppConfig struct {
+	cfgs []*cfgFile
+}
+
+// NewAppConfig parses config files as per:
+// https://docs.ota.here.com/ota-client/latest/aktualizr-config-options.html#_how_toml_files_are_processed
+func NewAppConfig(configPaths []string) (*AppConfig, error) {
+	configsMap := make(map[string]string)
+
+	// Build up the files
+	for _, path := range configPaths {
+		if st, err := os.Stat(path); err == nil {
+			if st.IsDir() {
+				if entries, err := os.ReadDir(path); err == nil {
+					for _, entry := range entries {
+						if strings.HasSuffix(entry.Name(), ".toml") && entry.Type().IsRegular() {
+							configsMap[entry.Name()] = filepath.Join(path, entry.Name())
+						}
+					}
+				}
+			} else {
+				configsMap[st.Name()] = path
+			}
+		}
+	}
+
+	// Sort them alphabetically
+	keys := make([]*cfgFile, 0, len(configsMap))
+	for k, v := range configsMap {
+		keys = append(keys, &cfgFile{name: k, path: v})
+	}
+	sort.Slice(keys[:], func(i, j int) bool {
+		return keys[i].name < keys[j].name
+	})
+
+	var err error
+	for _, cfg := range keys {
+		cfg.tree, err = toml.LoadFile(cfg.path)
```

Comment:
Is this how the merge occurs, through the assignment cfg.tree (current tree) = toml.LoadFile(cfg.path) (specific file tree)?

Reply:
Ah, no merge, sorry.